taichi/ui/backends/vulkan/renderables/particles.cpp

```diff
@@ -69,15 +69,32 @@ void Particles::init_particles(AppContext *app_context,
       true,
       app_context->config.package_path + "/shaders/Particles_vk_vert.spv",
       app_context->config.package_path + "/shaders/Particles_vk_frag.spv",
-      TopologyType::Points,
+      TopologyType::Triangles,  // We use two triangles to draw out a quad
       PolygonMode::Fill,
       vbo_attrs,
+      true  // point instancing
   };
 
   Renderable::init(config, app_context);
   Renderable::init_render_resources();
 }
 
+void Particles::record_this_frame_commands(CommandList *command_list) {
+  command_list->bind_pipeline(pipeline_.get());
+  command_list->bind_raster_resources(raster_state_.get());
+  command_list->bind_shader_resources(resource_set_.get());
+
+  // We draw num_particles * 6, 6 forms a quad
+  if (indexed_) {
+    command_list->draw_indexed_instance(6, config_.draw_index_count,
```

Comment:
hmmmmmmm.... why would "indexed" particles make any sense? I need to double check whether GGUI API provides indexed version for particles.